lib/ansible/modules/cloud/vmware/vmware_datastore_host.py

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/python
+
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vmware_datastore_host
+short_description: Add/remove datastore on ESXi host
+description:
+- This module can be used to mount/umount on datastore on ESXi host.
+- Currently, this module only support NFS datastore type but it's planned to add vmfs support.
```

Comment:
looking by your code, vmfs seems to be supported…

Reply:
You're right : it was a restriction from a draft version only. fixing.